src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -548,6 +550,14 @@ private static DOMParser createDOMParser() throws SamlException {
                   "Cannot disable comments parsing to mitigate https://www.kb.cert.org/vuls/id/475445",
                   ex);
             }
+
+            try {
+              setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+              setFeature(XERCES_FEATURE_PREFIX + DISALLOW_DOCTYPE_DECL_FEATURE, true);
```

Comment:
Based on [org.apache.xerces.impl.Constants documentation](https://xerces.apache.org/xerces2-j/javadocs/xerces2/org/apache/xerces/impl/Constants.html), the value of `XERCES_FEATURE_PREFIX` is `"http://apache.org/xml/features/"` and the value of `DISALLOW_DOCTYPE_DECL_FEATURE` is `"disallow-doctype-decl"`. So we are basically calling `setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);` twice.

Is that expected? 🙂

Reply:
I............... guess that's my idea of "fix it twice" 🤦‍♂